### PR TITLE
Pin dask-expr; realign A/B envs

### PR DESCRIPTION
**Pull request by crusaderky**

### CODE REVIEW

File: ci/environment.yml [hunk 1]
```diff
@@ -1,19 +1,14 @@
 channels:
   - conda-forge
 dependencies:
-  - conda
```
Comment (crusaderky) [inline ci/environment.yml:4]: redundant with environment-test.yml

File: ci/environment.yml [hunk 2]
```diff
@@ -1,19 +1,14 @@
...
-  - python-duckdb ==0.9.1
-  - polars >=0.19.3,<=0.19.8  # 0.19.3 on Windows, 0.19.8 on Linux and MacOS
   - pip
   - coiled >=0.2.54
   - numpy ==1.24.4
   - pandas ==2.1.1
   - dask ==2023.9.2
   - distributed ==2023.9.2
-  - dask-expr
+  - dask-expr ==0.1.8
```
Comment (crusaderky) [inline ci/environment.yml:11]: Note that both before and after this is overridden by environment-git-tip.yml. This change is for the benefit of A/B testing.

**Review by milesgranger — APPROVED** [review]
Thank you!

File: ci/environment.yml (hunk 2 above)
Comment (phofl) [reply ci/environment.yml:11]: So nightly will consistently run with main?

Comment (crusaderky) [reply ci/environment.yml:11]: Yes, both nightly and PRs use the overlay of
1. python={3.9|3.10|3.11} depending on matrix in `.github/workflows/tests.yml`
2. `ci/environment.yml`
3. in a single matrix configuration, `ci/environment-snowflake.yml`
4. `ci/environment-git-tip.yml`
5. `ci/environment-test.yml`

Whereas A/B tests use
1. one of `AB_environments/AB_*.conda.yml`
2. `ci/environment-test.yml`

**Pull request merged**